### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by hjeong12**

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/**
+ * events.js
+ */
+var utils = require('./utils'),
+  slice = Array.prototype.slice;
+
+//define entire events
+var allEvents = ['bidRequested','bidResponse','bidWon','bidTimeout'];
```
Comment (nickjacob) [inline src/events.js]:
```javascript
var constants = require('./constants'),
      allEvents = utils._map(constants.EVENTS, function (v){ return v; });
```

^ so we don't have to maintain list?

Comment (mkendall07) [inline src/events.js]: Yep good idea. I wanted to replace these but forgot to. Thanks

File: src/ga.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	if (dollars) {
+		return Math.floor(dollars * 100);
+	}
+	return 0;
+}
+
+function getLoadTimeDistribution(time) {
+	var distribution;
+	if (time >= 0 && time < 200) {
+		distribution = '0-200ms';
```
Comment (nickjacob) [inline src/ga.js]:
```javascript
var roundedTime = Math.floor(time / 100) * 100,
  distribution;

if (roundedTime < 200) {
  distribution = '0-200ms';
} else if (roundedTime > MAX_TIME) {
  distribution = MAX_TIME + 'ms above';
} else {
  distribution = (roundedTime + '-' + (roundedTime+100) + 'ms');
}
```
is this same as ^? Just hoping to reduce branches/ops for higher values

File: src/ga.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		distribution = '$2-2.5';
+	} else if (cpm >= 2.5 && cpm < 3) {
+		distribution = '$2.5-3';
+	} else if (cpm >= 3 && cpm < 4) {
+		distribution = '$3-4';
+	} else if (cpm >= 4 && cpm < 6) {
+		distribution = '$4-6';
+	} else if (cpm >= 6 && cpm < 8) {
+		distribution = '$6-8';
+	} else if (cpm >= 8) {
```
Comment (nickjacob) [inline src/ga.js]: I guess maybe for above also -- maybe have function

```javascript
function roundTo(num, increment) {
  return Math.floor(num / increment) * increment;
}
```

File: src/prebid.js [hunk 4]
```diff
@@ -55,8 +59,14 @@ pbjs.que.push = function(cmd) {
...
 	for (var i = 0; i < pbjs.que.length; i++) {
 		if (typeof pbjs.que[i].called === objectType_undefined) {
-			pbjs.que[i].call();
-			pbjs.que[i].called = true;
+			try{
+				pbjs.que[i].call();
+				pbjs.que[i].called = true;
+			}
+			catch(e){
+				utils.logError('Error processing command :', 'prebid.js', e);
```
Comment (nickjacob) [inline src/prebid.js]: slightly unrelated to this PR, but what do you think about events (via `events.emit`) on caught errors?

File: src/ga.js (hunk 3 above)
Comment (mkendall07) [inline src/ga.js]: Yeah I think so. I can test it easily. Thanks

File: src/utils.js [hunk 5]
```diff
@@ -387,7 +385,21 @@ exports._each = function(object, fn) {
       for (; k < l; k++) fn(object[k], k, object);
     } else {
       for (k in object) {
-        if (hasOwnProperty.call(object, k)) fn(object[k], k, object);
+        if (hasOwnProperty.call(object, k)) fn.call(this, object[k], k);
       }
     }
   };
+
+exports.contains = function(a, obj) {
```
Comment (nickjacob) [inline src/utils.js]: could call `indexOf` if available & cast to bool -- e.g.

```javascript
if (this.isFn(a.indexOf)) {
  return a.indexOf(obj) !== -1;
}
```

File: src/prebid.js (hunk 4 above)
Comment (mkendall07) [inline src/prebid.js]: It doesn't hurt if we propagate it along I guess. What would be a use case for an internal error like this? Would you want to expose the handler publicly?

**Comment by nickjacob** [comment]
this is going to be huge for ops

File: src/prebid.js (hunk 4 above)
Comment (nickjacob) [inline src/prebid.js]: just to send to an error reporting service (e.g., Honeybadger, Newrelic) -- I know GA also supports exception logging

**Comment by mkendall07** [comment]
Thanks for the review @nickjacob

**Pull request merged**